framework/src/org/checkerframework/framework/flow/CFAbstractValue.java

```diff
@@ -336,43 +344,53 @@ public V leastUpperBound(/*@Nullable*/ V other) {
                         other.getUnderlyingType(),
                         this.getAnnotations(),
                         other.getAnnotations(),
-                        lub);
+                        lub,
+                        shouldWiden);
         lubVisitor.visit();
         return analysis.createAbstractValue(lub, lubTypeMirror);
     }
 
     class LubVisitor extends AnnotationSetAndTypeMirrorVisitor {
-
         Set<AnnotationMirror> lubSet;
+        boolean widen;
 
         public LubVisitor(
                 TypeMirror result,
                 TypeMirror aTypeMirror,
                 TypeMirror bTypeMirror,
                 Set<AnnotationMirror> aSet,
                 Set<AnnotationMirror> bSet,
-                Set<AnnotationMirror> lubSet) {
+                Set<AnnotationMirror> lubSet,
+                boolean widen) {
             super(result, aTypeMirror, bTypeMirror, aSet, bSet);
             this.lubSet = lubSet;
+            this.widen = widen;
+        }
+
+        private AnnotationMirror lub(AnnotationMirror a, AnnotationMirror b) {
```

Comment:
In this class you introduced a helper method. Above I suggest that you do the same. The name `lub` isn't good, as it doesn't always perform `lub`.